### ISSUE TITLE
Paramètre hidenodata

### ISSUE THREAD

**Issue by bchartier** [geo2france/cog-tips#1]
J'ai l'impression que le fait de mettre le paramètre -hidenodata est une ânerie.
Il faut juste mettre -addalpha dans les commandes avec gdalbuildvrt.
Désolé pour l'erreur.

**Comment by Doctor-Who** [comment]
Salut Benjamin,

Ce n'est pas ton erreur, cela provient bien des commandes de la première version du guide.
J'ai suivi les indications de la doc officielle : https://gdal.org/programs/gdalbuildvrt.html

> **-hidenodata**
> Even if any band contains nodata value, giving this option makes the VRT band not report the NoData. Useful when you want to control the background color of the dataset. By using along with the -addalpha option, you can prepare a dataset which doesn't report nodata value but is transparent in areas with no data.

Il faut bien combiner les deux pour masquer le nodata comme transparence. Sinon ce sera une valeur noire ou blanche selon les préférences définies dans ton QGIS par exemple.

**Comment by bchartier** [comment]
J'ai bien lu cela mais ça a l'air contradictoire avec les tests faits ces derniers jours pour DataGrandEst.
Je vais reprendre mes tests pour voir ce qui cloche.
Laisse ouvert le ticket le temps d'éclaircir la situation.

**Comment by bchartier** [comment]
J'ai fait des tests et j'y vois un peu plus clair.
J'ai rencontré un cas où il ne faut pas ajouter -hidenodata mais c'est parce qu'il ne faut pas ajouter -addalpha. Il y a deux cas que j'ai identifiés :
1. le cas où il y a déjà un canal alpha dans les sources
2. le cas où on ne veut pas mettre de transparence et conserver les zones sans données totalement noires comme c'est le cas dans la source (pour continuer à gérer la transparence du côté de GeoServer ou QGIS par exemple en lui indiquant le que noir parfait doit être transparent).

Par contre, je ne m'explique la différence dans la première commande de création de VRT entre la version Linux et Windows
https://github.com/geo2france/cog-tips/blob/main/README.md?plain=1#L20
Dans la version Linux on a -addalpha mais pas -hidenodata.
Il y a une raison ?

**Comment by Doctor-Who** [comment]
Ok pour les 2 cas cités. Je comprends.
Il peut y avoir des effets d'artefacts parce qu'il peut y avoir du noir "pur" aussi dans l'image. Cela se produit avec les images Pléiades par exemple.
Alors, il faudra découper le contour de l'image.

Pour la différence, c'est une erreur de copier-coller, je vais corriger tout de suite.
Pour du MNT provenant d'une donnée de type GRID (asc, etc), il n'y a pas besoin généralement avec la BDAlti de gérer le nodata.

**Comment by bchartier** [comment]
> Il peut y avoir des effets d'artefacts parce qu'il peut y avoir du noir "pur" aussi dans l'image. Cela se produit avec les images Pléiades par exemple.

Oui. Dans le cas que j'avais en tête ce n'est pas le cas mais j'ai bien conscience de cela.

[J'essaie depuis hier soir de mettre au clair des règles sur la gestion des zones sans données dans différents cas de figure. Je me rends compte que ce n'est pas facile à expliquer au personnes auxquelles je dois faire un transfert de compétence. Je crois qu'on aurais intérêt à faire un doc pédagogique assez complet à diffuser largement. Je te tiens au courant.]
